File: BusCatalog.Api/BusCatalog.Api/Domain/Lines/Line.cs
using BusCatalog.Api.Domain.Vehicles;

namespace BusCatalog.Api.Domain.Lines;

public record Line
{
    public int Id { get; set; }
    public required string Identification { get; set; }
    public required string Fromwards { get; set; }
    public required string Towards { get; set; }
    public int DeparturesPerDay { get; set; }
    public virtual List<Vehicle> Vehicles { get; set; } = [];

    public void AddVehicles(IEnumerable<Vehicle> vehicles) =>
        Vehicles.AddRange(vehicles);
}

File: BusCatalog.Scraper/src/main/java/br/com/marcomvidal/buscatalog/scraper/line/Line.java
package br.com.marcomvidal.buscatalog.scraper.line;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class Line {
    private String identification;
    private String fromwards;
    private String towards;
    private int departuresPerDay;
    private List<String> vehicles;

    public Line(
        String identification,
        String fromwards,
        String towards,
        int departuresPerDay) {
        this.identification = identification;
        this.fromwards = fromwards;
        this.towards = towards;
        this.departuresPerDay = departuresPerDay;
    }

    public Line withVehicles(List<String> vehicles) {
        this.vehicles = vehicles;

        return this;
    }
}
